project/src/main/scala/ScapegoatInspectionsGenerator.scala

```diff
@@ -49,33 +45,24 @@ object ScapegoatInspectionsGenerator {
         "ScapegoatInspections.scala"
       )
 
-    val allScapegoatInspections = extractInspections()
-    val stringifiedScapegoatIsnpections = stringifyInspections(allScapegoatInspections)
-    val transformed = fillTemplate(templateFile.parse[Source].get, stringifiedScapegoatIsnpections)
+    val allScapegoatInspections: List[(String, Inspection)] = extractInspections()
+    val stringifiedScapegoatIsnpections: List[String] = stringifyInspections(allScapegoatInspections)
+    val transformed: Tree = fillTemplate(templateFile.parse[Source].get, stringifiedScapegoatIsnpections)
 
-    val scapegoatInspectionsFile = (sourceManaged in Compile).value / "scapegoat" / "inspections.scala"
+    val scapegoatInspectionsFile: File = (sourceManaged in Compile).value / "scapegoat" / "inspections.scala"
     IO.write(scapegoatInspectionsFile, transformed.syntax)
 
     Seq(scapegoatInspectionsFile)
   }
 
-  /** Returns all scapegoat inspections, except the ones that should be ignored */
+  /**
+   * Returns all scapegoat inspections, except the ones that should be ignored
+   */
   def extractInspections(): List[(String, Inspection)] =
-    // We need to override the scanner class loader,
-    // so it can cast the loaded classes
-    // from class[_] to class[Inspection]
-    new ClassGraph()
-      .overrideClassLoaders(classOf[Inspection].getClassLoader)
-      .whitelistPackages("com.sksamuel.scapegoat.inspections")
-      .scan()
-      .getSubclasses("com.sksamuel.scapegoat.Inspection")
-      .loadClasses(classOf[Inspection])
-      .asScala
-      .toList
-      .collect {
-        case clazz if !BlacklistedInspections.contains(clazz.getName) =>
-          (clazz.getName, clazz.newInstance())
-      }
+    ScapegoatConfig.inspections.collect {
```

Comment:
> Following above comment, this can be a simple `map`.

👍 

> Also, do we need it to be a `List`? or are we happy with it being a `Seq`?

I generally tend to go for a `Seq` as a return type, which is a trait and it makes it easier to use any subtype depending on the needs. I'm not sure why we used a `List` here, but I'm not too fussed about it since `List` is a default implementation of `Seq` and I don't think we have the need to convert this to an indexed sequence anywhere.

Reply:
I used to go for `Seq` too, but a few days ago I found [this](https://gist.github.com/djspiewak/2ae2570c8856037a7738#generic-interfaces) and I must admit that he got a point, specially with the complexity.
Anyway, in this case I'm ok with it being a Seq since **ScapegoatConfig.inspections**  is a `Seq` too, and it will be a `List` in runtime, and we aren't exposing it.